ludwig/features/category_feature.py

```diff
@@ -137,13 +141,26 @@ def get_feature_meta(column, preprocessing_parameters, backend):
 
     @staticmethod
     def feature_data(column, metadata):
-        return column.map(
-            lambda x: (
-                metadata["str2idx"][x.strip()]
-                if x.strip() in metadata["str2idx"]
-                else metadata["str2idx"][UNKNOWN_SYMBOL]
```

Comment:
Let's keep this code for the `else` case, since we only care about logging a warning when we are using the fallback symbol when the `UNKNOWN_SYMBOL` is not in the metadata, yet we have found that we need to use the fallback symbol.